src/shadowbox/server/manager_service.ts

```diff
@@ -599,6 +608,41 @@ export class ShadowsocksManagerService {
     }
   }
 
+  async getServerMetrics(req: RequestType, res: ResponseType, next: restify.Next) {
+    try {
+      logging.debug(`getServerMetrics request ${JSON.stringify(req.params)}`);
+
+      if (!req.query?.since) {
+        return next(
+          new restifyErrors.MissingParameterError({statusCode: 400}, 'Parameter `since` is missing')
+        );
+      }
+
+      const timestamp = new Date(req.query.since as string);
+      const isIsoTimestamp =
+        !isNaN(timestamp.getTime()) && timestamp.toISOString() === req.query.since;
+
+      if (!isIsoTimestamp) {
+        return next(
+          new restifyErrors.InvalidArgumentError(
+            {statusCode: 400},
+            `'since' parameter must be an ISO timestamp. Got ${req.query.since}`
+          )
+        );
+      }
+
+      const hours = Math.floor((new Date().getTime() - timestamp.getTime()) / (1000 * 60 * 60));
```

Comment:
It looks like we are jumping through hoops to convert a timestamp to a duration.
Also, I believe the common use case will be to specify a duration.

Let's change the API to take a duration instead.

We can pass it in hours, or use something more user friendly like "30d", as [done in Go](https://pkg.go.dev/time#ParseDuration) and Cloudflare Radar: https://radar.cloudflare.com/?dateRange=12w